packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -978,35 +981,45 @@ class ProjectFileInvalidator {
     // Remove any packages which are derived from the pub cache.
     for (String packageName in packageMap.keys.toList()) {
       final String path = packageMap[packageName].path;
-      if ((platform.isWindows && path.contains(_pubCachePathWindows))
-          || path.contains(_pubCachePathLinuxAndWindows)) {
+      if (_shouldSkip(path)) {
         packageMap.remove(packageName);
       }
     }
   }
 
   List<String> findInvalidated() {
+    final List<String> invalidatedFiles = <String>[];
     final File packagesFile = fs.file(_packagesPath);
     if (packagesFile.existsSync()) {
       final int newPackagesUpdateTime = packagesFile.statSync().modified.millisecondsSinceEpoch;
       // Hot reloading with an updated package will often times kill a non-trivial
       // appliction. This _might_ work, given certain application size and package
-      // constraints, so instead of exiting we print a warning so that the user has
-      // some hint on what went wrong.
+      // constraints.
       if (newPackagesUpdateTime > _packagesUpdateTime) {
-        printError('Warning: updated dependencies detected. The Flutter application will require a restart to safely use new packages.');
+        final Map<String, Uri> newPackageMap = PackageMap(_packagesPath).map;
+        final Set<String> newPackages = Set<String>.from(newPackageMap.keys);
+        final Set<String> difference = newPackages.difference(_loadedPackages);
```

Comment:
Does `_loadedPackages` have to be instance variable or it just a local `Set<String>.from(_packageMap.keys)`?

Reply:
I store it as an instance variable before I remove the dev dependencies and pub cache from the packageMap

Reply:
but does _packageMap content changes outside of this function? It looks as if _loadedPackages is just content of the _packageMap before it is updated at line 1010 below in this function.